### PR TITLE
Drop connection after the defined retries and timeouts.  Closing issue #120

### PR DESCRIPTION
**Pull request by cr4pt0**
"peeking empty" happens when we connect to node with incorrect overlay_version:

there is new version that works good without "peeking empty" error: 

`let node_info = NodeInfo::new(19, 25, 23, "v19.5.0".to_string(), network);`

this is the old one that we stuck forever with "peeking empty" error:

`let node_info = NodeInfo::new(19, 21, 19, "v19.1.0".to_string(), network);`

There is no chance that we we will receive any message once we go to this case statement. 

```
Ok(Ok(0)) => {
     log::error!("peeking empty");
}
```

introduce retry_read variable. increment each time received zero size

Closing https://github.com/pendulum-chain/spacewalk/issues/120

### CODE REVIEW

**Review by ebma — COMMENTED** [review]
Thanks a lot for the description of the PR, and I also think the code is better now. 👍 

But just for some understanding before we merge this: so you are saying that once we get into this `Ok(Ok(0)) => {` match branch we will stay in that forever, ie while testing you experienced that you never got another message afterwards? Is this always the case?

Because if that's true we should maybe even think about immediately returning an error since there is no reason to retry if it will stay the same forever.

**Comment by cr4pt0** [comment]
> 

Yes. Exactly. Once we receive  and went into `Ok(Ok(0)) => {}`  i never receive something new.
Of cause we can return error immediately and never go this statement again.
But I pepefer to log banch a log just to have attention that it happen several time.
Because to return the error at the first time it is a really big changes for such an important component like stellar-relay receiver.

**Comment by ebma** [comment]
But is it a big change to return the error immediately? I don't see why it would be. 
However, since I am not sure what the inner workings of the stellar-overlay are in this regard, it would be fine for me to keep it the way you did it. If we encounter issues with that in the future, the error message should point us into the right direction.

**Review by ebma — APPROVED** [review]

**Comment by cr4pt0** [comment]
> But is it a big change to return the error immediately? I don't see why it would be. However, since I am not sure what the inner workings of the stellar-overlay are in this regard, it would be fine for me to keep it the way you did it. If we encounter issues with that in the future, the error message should point us into the right direction.

It is not a big changes to return error immediately but it is a important changes in perspective new functionality.

**Pull request merged**
